l1-contracts/contracts/bridge/asset-tracker/GWAssetTracker.sol

```diff
@@ -641,6 +641,9 @@ contract GWAssetTracker is AssetTrackerBase, IGWAssetTracker {
         // We save the chainBalance for the previous migration number so that the chain balance can be migrated back to GW in case it was not migrated.
         // Note, that for this logic to be correct, we need to ensure that `_chainMigrationNumber` is odd, i.e. the chain actually
         // actively settles on top of Gateway.
+        // We can safely decrement the _chainMigrationNumber by one, since at the moment we have an invariant that all chains start on L1,
+        // and have to be migrated to GW, so _chainMigrationNumber will be positive. That includes chains that were on GW before v30, we will
+        // migrate them to L1 before v30 upgrade, and they will get migrated to GW after.
```

Comment:
Can you also add that all L1->GW->L2 txs are processed on the chain after the L1->GW migration is successfully confirmed on L1? This has not been merged yet, but its worth mentioning